hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
The 10th percentile didn't seem to work (on this data set, at least). I'm thinking it could be made more robust if the y limit is set as some function of how many spectra are in the image?

I have to run to a meeting, but here is the code that I've been testing with:

```python
for i in ['scotts', 'freedman', 100]:
    sig = s
    navigation_mask = None
    dc = sig.data
    der = np.abs(np.diff(dc, 1, -1))
    n = ((~navigation_mask).sum() if navigation_mask else
                 sig.axes_manager.navigation_size)
    if n < 200:
        tmp = signals.Signal(der)
        tmph = tmp.get_histogram(bins=i, range_bins=(0.9 * tmp.data.min(), 1.1 * tmp.data.max()))
        ylim = np.max(tmph.data) * 0.1
    else:
        tmp=signals.Signal(np.ravel(der.max(-1)))
        tmph = tmp.get_histogram(bins=i)
        ylim = np.max(tmph.data) * 0.1
    tmph.metadata.General.title = "LogY histogram with bins = " + str(i)
    tmph.plot()
    plt.xlabel('Threshold')
    plt.ylabel('Counts')
    plt.gca().set_title(tmph.metadata.General.title)
    plt.gca().set_ylim(0,(sig.axes_manager.navigation_size * sig.axes_manager.signal_size) * 0.000001)
   # plt.gca().set_yscale('log')
  #  plt.gcf().savefig('C:\\tmp\\2-reg_log_' + tmph.metadata.General.title + '.png')

```